### PR TITLE
add Dockerfiles for container image that includes a (working) fuse-overlayfs

### PR DESCRIPTION
**Pull request by boegel**
Resulting container images are available through Docker Hub, see https://hub.docker.com/r/eessi/fuse-overlay/tags

To use:

```bash
# create temporary directory
export EESSI_TMPDIR=/tmp/$USER
mkdir -p $EESSI_TMPDIR

# create scratch dirs for $HOME, overlayfs and CVMFS
mkdir -p $EESSI_TMPDIR/{home,overlay-upper,overlay-work}
mkdir -p $EESSI_TMPDIR/{var-lib-cvmfs,var-run-cvmfs}

# specify Singularity configuration
export SINGULARITY_HOME="$EESSI_TMPDIR/home:/home/$USER"
export SINGULARITY_BIND="$EESSI_TMPDIR/var-run-cvmfs:/var/run/cvmfs,$EESSI_TMPDIR/var-lib-cvmfs:/var/lib/cvmfs"
export SINGULARITY_CACHEDIR=$EESSI_TMPDIR/singularity_cache

# values to pass to --fusemount in singularity command
export EESSI_CONFIG="container:cvmfs2 cvmfs-config.eessi-hpc.org /cvmfs/cvmfs-config.eessi-hpc.org"
export EESSI_PILOT_READONLY="container:cvmfs2 pilot.eessi-hpc.org /cvmfs_ro/pilot.eessi-hpc.org"
export EESSI_PILOT_WRITABLE_OVERLAY="container:fuse-overlayfs -o lowerdir=/cvmfs_ro/pilot.eessi-hpc.org -o upperdir=$EESSI_TMPDIR/overlay-upper -o workdir=$EESSI_TMPDIR/overlay-work /cvmfs/pilot.eessi-hpc.org"

# start shell in container
singularity shell --fusemount "$EESSI_CONFIG" --fusemount "$EESSI_PILOT_READONLY" --fusemount "$EESSI_PILOT_WRITABLE_OVERLAY" docker://eessi/fuse-overlay:debian10-$(uname -m)
```

### CODE REVIEW

File: Dockerfile.fuse-overlay-debian10-aarch64 [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+RUN wget https://ecsft.cern.ch/dist/cvmfs/cvmfs-release/cvmfs-release-latest_all.deb &&  \
+  dpkg -i cvmfs-release-latest_all.deb && \
+  rm -f cvmfs-release-latest_all.deb
+
+RUN apt-get install -y debhelper autotools-dev cmake cpio libcap-dev libssl-dev pkg-config libattr1-dev patch \
+  python-dev python-setuptools unzip uuid-dev valgrind libz-dev gawk psmisc autofs curl attr zlib1g gdb uuid-dev \
+  uuid adduser autofs psmisc curl attr openssl libcap2 libcap2-bin lsof rsync jq usbutils sqlite3 sudo \
+  fuse libfuse-dev libfuse2 libfuse3-dev
+
+# build CervVM-FS from source (no aarch64 Debian packages available)
```
Comment (bedroge) [inline Dockerfile.fuse-overlay-debian10-aarch64:15]:
```suggestion
# build CernVM-FS from source (no aarch64 Debian packages available)
```

File: Dockerfile.fuse-overlay-debian10-aarch64 [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+  make -j $(nproc) && \
+  sudo make install && \
+  cd / && \
+  rm -r cvmfs*2.7.5*
+
+RUN wget https://github.com/EESSI/filesystem-layer/releases/download/v0.2.3/cvmfs-config-eessi_0.2.3_all.deb && \
+  dpkg -i cvmfs-config-eessi_0.2.3_all.deb && \
+  rm -f cvmfs-config-eessi_0.2.3_all.deb
+
+# install fuse3 to Singularity's --fusemount works
```
Comment (bedroge) [inline Dockerfile.fuse-overlay-debian10-aarch64:32]:
```suggestion
# install fuse3 so Singularity's --fusemount works
```

**Review by bedroge — CHANGES_REQUESTED** [review]

**Comment by boegel** [comment]
I should probably update the README too, or the docs, with instructions how to use these containers...

**Comment by bedroge** [comment]
That can probably be part of some software build manual (cleaned up version of your notes/document)? I'll merge this for now, and we can add docs later / in a separate PR.

**Review by bedroge — APPROVED** [review]

**Pull request merged**